packages/dds/merge-tree/src/perspective.ts

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { type MergeTree } from "./mergeTree.js";
+import { LeafAction, backwardExcursion, forwardExcursion } from "./mergeTreeNodeWalk.js";
+import { isSegmentPresent, type ISegment } from "./mergeTreeNodes.js";
+
+/**
+ * Defines a side relative to an element in a sequence.
+ *
+ * @public
+ */
+export enum Side {
+	Before = 0,
+	After = 1,
+}
+
+/**
+ * A place between two elements in a sequence.
+ * A `pos` of `undefined` means the extents of the sequence.
+ * Before undefined is the end of the sequence, After undefined is the start of the sequence.
+ *
+ * @public
+ */
+export interface Place {
+	/**
+	 * The index of the reference sibling measured from the start of the sequence.
+	 */
+	readonly pos: number | undefined;
+	readonly side: Side;
+}
+
+/**
+ * Provides a view of a MergeTree from the perspective of a specific client at a specific sequence number.
+ *
+ * @public
+ */
+export interface Perspective {
+	nextSegment(segment: ISegment, forward?: boolean): ISegment;
+	previousSegment(segment: ISegment): ISegment;
+}
+
+/**
+ * Implementation of {@link Perspective}.
+ * See {@link Client.createPerspective}.
+ *
+ * @internal
+ */
+export class PerspectiveImpl implements Perspective {
+	/**
+	 * @param _mergeTree - The {@link MergeTree} to view.
+	 * @param _sequenceNumber - The latest sequence number to include in the view.
+	 * @param _clientId - The client ID to view from.
+	 * @param _localSequenceNumber - The latest local sequence number to include in the view.
+	 */
+	public constructor(
+		private readonly _mergeTree: MergeTree,
```

Comment:
this dependency ends up being circular. could we just take the parts of merge tree we need? maybe even in a way the a the merge tree itself satisfies like `{endOfTree, startOfTree}`

Reply:
The plan is for Perspective to eventually wrap more things, such as `MergeTree.getContainingSegment` and `MergeTree.getPosition`. I would rather not make an `IMergeTree` interface just for this.

The circular import shouldn't matter so long as neither file has side-effects on import?

Reply:
Additionally, the `./mergeTree.js` import is type-only. It does not get imported at runtime.